src/mbgl/style/expression/interpolate.cpp

```diff
@@ -216,6 +216,33 @@ std::vector<optional<Value>> InterpolateBase::possibleOutputs() const {
     return result;
 }
 
+template <typename T>
+mbgl::Value Interpolate<T>::serialize() const {
+    std::vector<mbgl::Value> serialized;
+    serialized.emplace_back(getOperator());
+    
+    interpolator.match(
+        [&](const ExponentialInterpolator& exponential) {
+            serialized.emplace_back(std::vector<mbgl::Value>{{ std::string("exponential"), exponential.base }});
+        },
+        [&](const CubicBezierInterpolator& cubicBezier) {
+            static const std::string cubicBezierTag("cubic-bezier");
+            auto p1 = cubicBezier.ub.getP1();
+            auto p2 = cubicBezier.ub.getP2();
+            serialized.emplace_back(std::vector<mbgl::Value>{{ cubicBezierTag, p1.first, p1.second, p2.first, p2.second }});
+        },
+        [&](const auto&) {
+            assert(false);
+        }
```

Comment:
This noop branch shouldn't be necessary, because the specific cases above cover every member type of the variant.